src/init.cpp

```diff
@@ -217,7 +217,7 @@ void Shutdown(NodeContext& node)
     node.banman.reset();
     node.addrman.reset();
 
-    if (node.mempool && node.mempool->IsLoaded() && node.args->GetArg("-persistmempool", DEFAULT_PERSIST_MEMPOOL)) {
+    if (node.mempool && node.mempool->IsLoaded() && node.args->GetIntArg("-persistmempool", DEFAULT_PERSIST_MEMPOOL)) {
```

Comment:
@MarcoFalke's #23061 changes this to `GetBoolArg`. Given the ACK there, I assume that is the more correct change? If that is going to be backported then it can be merged first and this PR rebased on top.

Reply:
(I would expect so too as `DEFAULT_PERSIST_MEMPOOL` is a boolean constants.)